data/milvus/v0/tasks.go

```diff
@@ -0,0 +1,627 @@
+	case "TANIMOTO":
+		metricType = entity.TANIMOTO
+	case "SUBSTRUCTURE":
+		metricType = entity.SUBSTRUCTURE
+	case "SUPERSTRUCTURE":
+		metricType = entity.SUPERSTRUCTURE
+	default:
+		return nil, fmt.Errorf("unsupported metric type")
+	}
+
+	var sp entity.SearchParam
+	switch inputStruct.SearchParameter {
+	case "FLAT":
+		sp = &entity.IndexFlatSearchParam{}
+	case "BIN_FLAT":
+		sp = &entity.IndexBinFlatSearchParam{}
+	case "DISKANN":
+		sp = &entity.IndexDISKANNSearchParam{}
+	case "GPU_IVF_FLAT":
+		sp = &entity.IndexGPUIvfFlatSearchParam{}
+	case "GPU_IVF_PQ":
+		sp = &entity.IndexGPUIvfPQSearchParam{}
+	case "HNSW":
+		sp = &entity.IndexHNSWSearchParam{}
+	case "IVF_FLAT":
+		sp = &entity.IndexIvfFlatSearchParam{}
+	case "IVF_PQ":
+		sp = &entity.IndexIvfPQSearchParam{}
+	case "IVF_SQ8":
+		sp = &entity.IndexIvfSQ8SearchParam{}
+	case "SCANN":
+		sp = &entity.IndexSCANNSearchParam{}
+	case "BIN_IVF_FLAT":
+		sp = &entity.IndexBinIvfFlatSearchParam{}
+	default:
+		return nil, fmt.Errorf("unsupported search parameter")
+	}
+
+	describe, err := e.client.DescribeCollection(context.Background(), collectionName)
+	if err != nil {
+		panic(err)
+	}
+
+	if fields == nil {
+		for _, field := range describe.Schema.Fields {
+			fields = append(fields, field.Name)
+		}
+	}
+
+	err = e.client.LoadCollection(context.Background(), collectionName, false)
+	if err != nil {
+		panic(err)
+	}
+
+	res, err := e.client.Search(context.Background(), collectionName, partitionName, filter, fields, []entity.Vector{vector}, "vector", metricType, topK, sp)
+	if err != nil {
+		panic(err)
+	}
+
+	var ids []string
+	var data []map[string]any
+	var vectors [][]float32
+	var metadata []map[string]any
+
+	for _, datum := range res {
+		rawVector, err := entityColumnToJSON(datum.Fields.GetColumn(vectorField))
+		if err != nil {
+			panic(err)
+		}
+
+		arrayVector := rawVector.([]float32)
+
+		for i := 0; i < len(arrayVector); i += len(vector) {
+			end := i + 2
+			if end > len(arrayVector) {
+				end = len(arrayVector)
+			}
+
+			vectors = append(vectors, arrayVector[i:end])
+		}
+
+		id, err := entityColumnToJSON(datum.IDs)
+		if err != nil {
+			panic(err)
+		}
+		idArray := toAnySlice(id)
+		for _, v := range idArray {
+			ids = append(ids, fmt.Sprintf("%v", v))
+		}
+
+		listField := make(map[string]any)
+		listField[vectorField] = vectors
+		for _, field := range datum.Fields {
+			if field.Name() != vectorField {
+				val, err := entityColumnToJSON(datum.Fields.GetColumn(field.Name()))
+				if err != nil {
+					panic(err)
+				}
+				listField[field.Name()] = val
+			}
+		}
+		sort.Slice(datum.Scores, func(i, j int) bool {
+			return datum.Scores[i] > datum.Scores[j]
+		})
+		listField["score"] = datum.Scores
+
+		for i := 0; i < len(ids); i++ {
+			metadatum := make(map[string]any)
+			datum := make(map[string]any)
+			for k, v := range listField {
+				arrayV := toAnySlice(v)
+				if k != "score" && k != describe.Schema.PKFieldName() {
```

Comment:
`if k != "score" && k != describe.Schema.PKFieldName()`
Could you explain this part a bit?